Assets/VRTK/Source/Scripts/Utilities/VRTK_SharedMethods.cs

```diff
@@ -472,6 +469,109 @@ public static Vector3 AxisDirection(int axisIndex, Transform givenTransform = nu
+        /// <returns>Returns `true` if the given value was successfully added to the list. Returns `false` if the given value already existed in the list and `forceAdd = false`.</returns>
+        public static bool AddListValue<TValue>(ref List<TValue> list, TValue value, bool forceAdd = false)
+        {
+            if (forceAdd || !list.Contains(value))
+            {
+                list.Add(value);
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// The RemoveListValue method attempts to remove an entry from the given list based on the given value.
+        /// </summary>
+        /// <typeparam name="TValue">The datatype for the list value.</typeparam>
+        /// <param name="list">The list to remove the value from.</param>
+        /// <param name="value">The value to remove.</param>
+        /// <returns>Returns `true` if the value was found and removed. Returns `false` if the value was not found and therefore not removed.</returns>
+        public static bool RemoveListValue<TValue>(ref List<TValue> list, TValue value)
+        {
+            if (list.Contains(value))
+            {
+                list.Remove(value);
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// The GetDictionaryValue method attempts to retrieve a value from a given dictionary for the given key. It removes the need for a double dictionary lookup to ensure the key is valid and has the option of also setting the missing key value to ensure the dictionary entry is valid.
+        /// </summary>
+        /// <typeparam name="TKey">The datatype for the dictionary key.</typeparam>
+        /// <typeparam name="TValue">The datatype for the dictionary value.</typeparam>
+        /// <param name="dictionary">The dictionary to retrieve the value from.</param>
+        /// <param name="key">The key to retrieve the value for.</param>
+        /// <param name="setMissingKey">If this is `true` and the given key is not present, then the dictionary value for the given key will be set to the `setValue` parameter. If this is `false` and the given key is not present then the `setValue` parameter will be returned as the value.</param>
+        /// <param name="setValue">The value to utilise when either setting the missing key (if `setMissingKey` is `true`) or the default value to return when no key is found (if `setMissingKey` is `false`).</param>
+        /// <returns>The found value for the given key in the given dictionary, or the default value if no key is found.</returns>
+        public static TValue GetDictionaryValue<TKey, TValue>(Dictionary<TKey, TValue> dictionary, TKey key, bool setMissingKey = false, TValue setValue = default(TValue))
+        {
+            if (dictionary == null)
+            {
+                return default(TValue);
```

Comment:
This should perhaps return a `defaultValue` which is supplied as a parameter instead.

Since this means the method has a lot parameters with confusing `setValue` vs. `defaultValue` I suppose splitting this method into two is a good idea, though I'm not sure what the use cases are exactly. E.g. do we need a way to set a different `defaultValue` compared to `setValue` at all?

Reply:
the `setValue` param could be used here as it is already defaulted to the `default` value anyway.

i dunno if it's accurate to say it has lots of confusing params, two ambiguous at best ;)

Reply:
I agree. Maybe just use that parameter then and add it to its documentation. (I can't think of a better name than `defaultValue` and `setDefaultValue` as a replacement for the existing parameter names.)